osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/SecretManagementServiceTest.java

```diff
@@ -145,8 +145,7 @@ public void testHasNewKey() {
     when(this.secretManagementClient.hasNewSecretRequest(same(messageMetadata), any()))
         .thenReturn(response);
     // EXECUTE
-    final boolean result =
-        this.testService.hasNewSecretOfType(messageMetadata, DEVICE_IDENTIFICATION, KEY_TYPE);
+    final boolean result = this.testService.hasNewSecret(messageMetadata, DEVICE_IDENTIFICATION);
```

Comment:
I would split this test into 2 tests: one where the authentication key is new and one where the encryption key is new.